backend/middlewares/RoleChecker.ts

```diff
@@ -0,0 +1,24 @@
+import { Response, Request, NextFunction } from 'express';
+import UserModel from '../models/UserModel';
+
+const RoleChecker = {
+  restrictTo(...roles: Array<string>) {
```

Comment:
maybe, instead of string Array, set tuple type with concrete roles. Like ['admin', 'user', 'moderator'].
Or global type Roles = ['admin', 'user', 'moderator'], in types.ts

Reply:
And where we use RoleChecker? Can't find in this PR